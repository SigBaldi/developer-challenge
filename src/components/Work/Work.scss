@import '../../styles/variables';

.Work {
  border: 1px solid $cloud;
  box-sizing: content-box;
  display: block;
  flex: 1 0 auto;
  height: 300px;
  margin: 0 0 -1px -1px;
  padding: 0 10px 20px;
  text-decoration: none;
  width: calc(50% - 22px);

  a,
  a:link,
  a:visited,
  a:hover,
  a:active {
    color: inherit;
  }

  pre {
    border: 1px solid $dark;
    color: $dark;
    float: right;
    padding: 2px 10px;
    vertical-align: middle;
  }
  
  img {
    height: 200px;
    object-fit: contain;
    width: 100%;
  }
}
