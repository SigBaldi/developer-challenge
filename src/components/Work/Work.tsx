import React from 'react';
import PriceTag from '../PriceTag';
import STYLES from './Work.scss';

const getClassName = (className: string) => STYLES[className] || 'UNKNOWN';
const parseTags = (tags: string[]): string => {
  if (tags.includes('Sold Out')) return 'Sold Out';
  if (tags.includes('Coming soon')) return 'Coming Soon';
  return '';
};

export interface IWork {
  id: number;
  image: string;
  price: number;
  tags: string[];
  title: string;
  url: string;
}

function Work(props: { work: IWork }): JSX.Element {
  const { image, price, tags, title, url } = props.work;
  const finalTag = parseTags(tags);
  return (
    <>
      <a href={url} className={getClassName('Work')}>
        {finalTag ? <pre>{finalTag}</pre> : <p>&nbsp;</p>}
        <img src={image} alt="Artwork" />
        <h3>{title}</h3>
        <PriceTag value={price} />
      </a>
    </>
  );
}

export default Work;
