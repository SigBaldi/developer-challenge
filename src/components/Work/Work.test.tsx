import React from 'react';
import TestRenderer from 'react-test-renderer';
import Work from './Work';

const mockWork = {
  id: 0,
  title: 'Dummy',
  tags: ['Sold Out'],
  image:
    '//cdn.shopify.com/s/files/1/0260/5524/5923/products/fightforadream_nav_8ecfb9fd-b156-4995-848a-6a5c84c9991a_500x.jpg?v=1617707578',
  url: 'https://shop.avantarte.com/products/dummy',
  price: 7500.0,
};

describe('Header', () => {
  it('should render correctly', () => {
    const tree = TestRenderer.create(<Work work={mockWork} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
