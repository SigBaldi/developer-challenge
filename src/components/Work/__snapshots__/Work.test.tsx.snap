// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render correctly 1`] = `
<a
  className="Work"
  href="https://shop.avantarte.com/products/dummy"
>
  <pre>
    Sold Out
  </pre>
  <img
    alt="Artwork"
    src="//cdn.shopify.com/s/files/1/0260/5524/5923/products/fightforadream_nav_8ecfb9fd-b156-4995-848a-6a5c84c9991a_500x.jpg?v=1617707578"
  />
  <h3>
    Dummy
  </h3>
  <span
    className="PriceTag"
  >
    â‚¬
    7500
  </span>
</a>
`;
